### PR TITLE
lesseon 2 done

### PR DESCRIPTION
**Pull request by Iman-feisali**
lesson 2 and some changes in lesson 1

### CODE REVIEW

File: src/Restaurant.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import React from "react";
 
+import Orders from "./Order";
```
Comment (Dedekind561) [inline src/Restaurant.js:3]:
```suggestion
import Order from "./Order";
```
You need to double check this one - you're importing a singular `Order` here :)

File: src/Restaurant.js [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 const Restaurant = () => {
-  const pizzas = 0;
   return (
     <div>
       <h3>Restaurant Orders</h3>
       <ul>
-        <li>
-          Pizzas: {pizzas} <button className="btn btn-primary">Add</button>
-        </li>
+        <Orders orderType={"Pizzas"} />
```
Comment (Dedekind561) [inline src/Restaurant.js:10]: 👍 - this looks good :) How could you render each of the Orders without rendering the Order component 3 times in a row. Think of how you do this with other components.

File: src/RestaurantButton.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+const RestaurantButton = props => {
```
Comment (Dedekind561) [inline src/RestaurantButton.js:3]: 👍

File: src/SearchResults.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { useState, useEffect } from "react";
+import TableHeader from "./TableHeader";
+import TableData from "./TableData";
+
+function SearchResults() {
+  useEffect(() => {
```
Comment (Dedekind561) [inline src/SearchResults.js:6]: This looks great to me :)

File: src/SearchResults.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    console.log("hello react");
+    fetch("https://cyf-react.glitch.me")
+      .then(res => {
+        return res.json();
+      })
+      .then(data => {
+        return setBookings(data);
+      });
+  }, []);
+  const [bookings, setBookings] = useState([]);
```
Comment (Dedekind561) [inline src/SearchResults.js:16]: Makes a little more sense to declare your state and the setter function at the top of your functional component - as you will need them later on

File: src/TableData.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import TableRow from "./TableRow";
+
+function TableData(props) {
+  return (
+    <tbody>
+      {props.results.map((item, index) => {
+        return <TableRow item={item} key={index} />;
```
Comment (Dedekind561) [inline src/TableData.js:8]: Good naming here in the map 😄

File: src/TableHeader.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+function TableHeader() {
+  return (
+    <thead>
+      <tr>
+        <th scope="col">#</th>
```
Comment (Dedekind561) [inline src/TableHeader.js:7]: Again here you could think about another way of rendering each of the columns without repeatedly writing out the `<th>...</th>` each time

File: src/TableRow.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <tr
+      id="table-row"
+      className={handleClick ? "tr-selected" : null}
+      onClick={() => {
+        setHandleClick(!handleClick);
+      }}
+    >
+      <th scope="row" />
+      <td>{props.item.id}</td>
```
Comment (Dedekind561) [inline src/TableRow.js:15]: Same as before with the rendering of individual `td` elements

File: src/TableRow.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React, { useState } from "react";
+import moment from "moment";
+
+const TableRow = props => {
```
Comment (Dedekind561) [inline src/TableRow.js:4]: Consider destructuring in here to access the `item` variable

File: src/TableRow.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  const [handleClick, setHandleClick] = useState(false);
```
Comment (Dedekind561) [inline src/TableRow.js:5]: `handleClick` sounds like the name of a function and not the name of a *boolean*

Comment (Dedekind561) [reply src/TableRow.js:5]: Hint: a good name for a variable would be something that starts with `is` or `has` something like that
Also think about what this state value represents

**Comment by Iman-feisali** [comment]
reopen for update

**Pull request closed without merging**